compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -766,13 +773,17 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                             VariantIdx::from_u32(variant_index)
                         } else {
                             dataful_variant
+                        };
+
+                        if is_relative {
+                            // A relative discriminant should only have been produced if this will fit.
+                            assert!(variant_index_relative <= discr_layout.size.unsigned_int_max());
+                            (Scalar::from_uint(variant_index_relative, discr_layout.size), variant)
+                        } else {
+                            (Scalar::from_uint(variant.as_u32(), discr_layout.size), variant)
```

Comment:
```suggestion
                            // The variant index directly serves as discriminant and is encoded in the tag.
                            (Scalar::from_uint(variant.as_u32(), discr_layout.size), variant)
```

Reply:
Why does this comment apply here? Seems like something someone should know before getting to the very last statement of this function.

Reply:
The other branch (`TagEncoding::Direct`) does '// Cast bits from tag layout to discriminant layout'. We have to explain why the same is not needed here in this branch.

In general, please don't just remove comments, even if you don't understand them. :) Instead ask what they are about so that we can improve them.

Reply:
Oh sorry; I didn't realized this was a comment I had removed.